hooks/boot/02-message-helpers.rb

```diff
@@ -38,13 +38,24 @@ def dev_new_install_message(kafo)
   end
 
   def failure_message
-    say "\n  <%= color('There were errors detected during install.', :bad) %>"
-    say "  Please address the errors and re-run the installer to ensure the system is properly configured."
-    say "  Failing to do so is likely to result in broken functionality."
+    say "\n<%= color('There were errors detected during installation.', :bad) %>"
+
+    if puppet_report&.failed_resources&.any?
+      puppet_report.failed_resources.each do |failed_resource|
+        say "\nResource '#{failed_resource}' failed. Logs:"
+        failed_resource.log_messages.each do |message|
+          say "  #{message}"
+        end
+      end
+      say "\n"
```

Comment:
```suggestion
      say "\n<%= color('#{puppet_report.failed_resources.length} errors were detected.', :bad) %>"
```

Include count of the number of errors detected.

Reply:
I like the number, but this then becomes very similar to the message before the errors are shown (line 41). Should I modify that line or drop it and keep this?

Reply:
Drop it and keep this.